### ISSUE TITLE
Publish to PVA from Bot Framework Composer Hangs and Never Completes

### ISSUE THREAD

**Issue by DS2019** [microsoft/BotFramework-Composer#5423]
## Describe the bug

Publishing to PVA just hangs and never completes 

## Version

32a4899

## Browser

- [ ] Electron distribution
- [X ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [X ] Windows
- [ ] Ubuntu

## To Reproduce

Choose a bot in Bot Framework Composer > go to Publish
Choose a Publish Profile (I've tried two)
Select Publish Selected Bots
Status just shows as a spinning wheel, nothing recorded in Log

## Expected behavior

I've never been able to publish a bot to PVA from Bot Framework Composer, but I would expect to see status "success" or similar and information about the publish in the log

## Screenshots

![PublishneverCompletes](https://user-images.githubusercontent.com/44073678/103386213-ab175800-4ab2-11eb-8798-5e22b89a2dc9.jpg)

**Comment by dmvtech** [comment]
Possibly related to #4659

**Comment by dmvtech** [comment]
I'm not even to get as far as the spinning wheel. Investigating.

**Comment by tonyanziano** [comment]
Hi @DS2019 ,

Do you happen to see anything in the network tab / console if you open DevTools with "Ctrl + Shift + I" before attempting the publish?

Any console errors or network responses higher than 400?

**Comment by DS2019** [comment]
Hi @tonyanziano - When I open up the publish page I get errors right away:
![PublishPage](https://user-images.githubusercontent.com/44073678/103684893-2b214000-4f41-11eb-83b3-b6113ecf9775.jpg)

**Comment by DS2019** [comment]
When I change to "Publish to PVA" nothing changes in the developer view. When I press to Publish, no changes:
![PUblishPVA](https://user-images.githubusercontent.com/44073678/103685159-95d27b80-4f41-11eb-8647-71c5c46fec11.jpg)

**Comment by tonyanziano** [comment]
I'm going to see if I can reproduce this

**Comment by tonyanziano** [comment]
Unfortunately I cannot reproduce this.

Could you please capture your network activity via a tool like Fiddler while you attempt to use the publishing flow?

It also might be helpful if you could send me the `publishTargets` property of your bot's `appsettings.json` file. You can find this in the `/settings/appsettings.json` file wherever you have saved your bot.

Feel free to send the exported network trace and the publish targets to my email at `toanzian@microsoft.com` and I'll take a look.

**Comment by tonyanziano** [comment]
After much back and forth with @DS2019 offline, I was able to uncover some odd behavior on the publish status page:

1. The front-end expects a 500 error code from the publisher to show an error and stop polling for status. This seems odd to me especially in the case that the publish operation did not error out, but instead attempted to publish and returned some other valid status code such as a 401 or 404, etc. This is what causes the infinite spinner.

2. There was also some sort of authentication problem when the opened bot project contained both an Azure publishing profile _and_ a PVA publishing profile. After removing the Azure publishing profile, @DS2019 was able to actually attempt a publish and get the correct status back from the back end.

This is not the expected behavior, and we will need to investigate further.

**Comment by Virtual-Josh** [comment]
@tonyanziano,

Have you had a chance to do the investigation? If so can you provide an update? If not, when will the investigation happen?

**Comment by tonyanziano** [comment]
Hey @Virtual-Josh ,

Regarding bullet 1, I have identified the issue in the front end code, but I don't have the cycles right now to fix it.

For the second bullet, I did some investigation on my own, and was unable to reproduce the issue using a similar setup to what @DS2019 had.

For now, I think resolving the underlying issue of bullet 1 will allow the UI to reflect the right publishing status and error information which will help us debug any further problems with publishing.
